### PR TITLE
Split UAA SecurityConfiguration in a distinct template

### PR DESCRIPTION
**Pull request by murdos**
And move security related templates in a dedicated template directory

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/files.js [hunk 1]
```diff
@@ -845,16 +855,6 @@ const serverFiles = {
                 },
             ],
         },
-        {
-            condition: generator => !generator.reactive && generator.authenticationType === 'oauth2',
```
Comment (murdos) [inline generators/server/files.js:849]: This one was redundant with https://github.com/jhipster/generator-jhipster/blob/9ce17c10f78c8d36a0d034280b1bde43f86d44d1/generators/server/files.js#L450, which means that for a non reactive oauth2 application, the file was written twice.

**Comment by mshima** [comment]
Great work on config files reorganisation.

Not convinced about the security folder though.
IMO it can be more confusing because we mimic destination package.

Same for database config, having `DatabaseConfiguration_cassandra.java.ejs` is already clean enough, don't need `database/DatabaseConfiguration_cassandra.java.ejs`.

**Comment by murdos** [comment]
I've reverted the move to the `security` subdirectory and applied the Filename_${option} naming pattern for SecurityConfiguration.java

**Review by mshima — APPROVED** [review]
LGTM

**Comment by pascalgrimaud** [comment]
I love what you're doing @murdos 
It will help a lot for v8, if we decide to use a clean architecture :)

**Comment by murdos** [comment]
@pascalgrimaud: thanks! 
These were a few breaking improvements that I had in mind and wanted to do in a major release, inspired by annoyances or pitfalls that I encountered while working on new features or framework upgrades in generator-jhipster.
I hope indeed that it'll easy maintenance and progress of JHipster 😄

**Pull request merged**